File: org.summer.view.widget/src/org/summer/view/widget/controls/primitives/ItemsChangedEventArgs.java
package org.summer.view.widget.controls.primitives;

import org.summer.view.widget.EventArgs;
import org.summer.view.widget.collection.NotifyCollectionChangedAction;

/// <summary>
/// The ItemsChanged event is raised by an ItemContainerGenerator to inform
/// layouts that the items collection has changed.
/// </summary> 
public class ItemsChangedEventArgs extends EventArgs
{ 
    //----------------------------------------------------- 
    //
    //  Constructors 
    //
    //-----------------------------------------------------

    /*internal*/public ItemsChangedEventArgs(NotifyCollectionChangedAction action, 
                                    GeneratorPosition position,
                                    GeneratorPosition oldPosition, 
                                    int itemCount, 
                                    int itemUICount)
    { 
        _action = action;
        _position = position;
        _oldPosition = oldPosition;
        _itemCount = itemCount; 
        _itemUICount = itemUICount;
    } 

    /*internal*/ public ItemsChangedEventArgs(NotifyCollectionChangedAction action,
                                    GeneratorPosition position, 
                                    int itemCount,
                                    int itemUICount) 

    { 
    	this(action, position, new GeneratorPosition(-1, 0), itemCount, itemUICount);
    }

    //------------------------------------------------------ 
    //
    //  Public Properties 
    //
    //-----------------------------------------------------

    /// <summary> What happened </summary> 
    public NotifyCollectionChangedAction Action { get { return _action; } }

    /// <summary> Where it happened </summary> 
    public GeneratorPosition Position       { get { return _position; } }

    /// <summary> Where it happened </summary>
    public GeneratorPosition OldPosition    { get { return _oldPosition; } }

    /// <summary> How many items were involved </summary> 
    public int ItemCount                    { get { return _itemCount; } }

    /// <summary> How many UI elements were involved </summary> 
    public int ItemUICount                  { get { return _itemUICount; } }


    //------------------------------------------------------
    //
    //  Private Fields 
    //
    //------------------------------------------------------ 

    NotifyCollectionChangedAction  _action;
    GeneratorPosition       _position; 
    GeneratorPosition       _oldPosition;
    int                     _itemCount;
    int                     _itemUICount;
} 

/// <summary> 
///     The delegate to use for handlers that receive ItemsChangedEventArgs. 
/// </summary>
//public delegate void ItemsChangedEventHandler(object sender, ItemsChangedEventArgs e); 

File: summer/WebContent/js/system/windows/controls/primitives/ItemsChangedEventArgs.js
/**
 * ItemsChangedEventArgs
 */

define(["dojo/_base/declare", "system/Type", "system/EventArgs"], function(declare, Type, EventArgs){
	var ItemsChangedEventArgs = declare("ItemsChangedEventArgs", EventArgs,{
		constructor:function(){
			if(arguments.length==5 ){
				this._action = arguments[0];
				this._position = arguments[1];
				this._oldPosition = arguments[2];
				this._itemCount = arguments[3];
				this._itemUICount = arguments[4];
				
			}else if(arguments.length==4 ){
				this._action = arguments[0];
				this._position = arguments[1];
				this._oldPosition = new GeneratorPosition(-1, 0);
				this._itemCount = arguments[2];
				this._itemUICount = arguments[3];
			}
		}
	});
	
	Object.defineProperties(ItemsChangedEventArgs.prototype,{

        /// <summary> What happened </summary> 
//        public NotifyCollectionChangedAction 
        Action: { get:function() { return this._action; } },
 
        /// <summary> Where it happened </summary> 
//        public GeneratorPosition 
        Position: { get:function() { return this._position; } },
 
        /// <summary> Where it happened </summary>
//        public GeneratorPosition 
        OldPosition: { get:function() { return this._oldPosition; } },

        /// <summary> How many items were involved </summary> 
//        public int 
        ItemCount: { get:function() { return this._itemCount; } },
 
        /// <summary> How many UI elements were involved </summary> 
//        public int 
        ItemUICount: { get:function() { return this._itemUICount; } }
	});
	
	ItemsChangedEventArgs.Type = new Type("ItemsChangedEventArgs", ItemsChangedEventArgs, [EventArgs.Type]);
	return ItemsChangedEventArgs;
});

